src/App.tsx

```diff
@@ -1,61 +1,128 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { ExtendedTodo, User } from './react-app-env';
+import { TodoList } from './components/TodoList';
+
+const findUserById = (users: User[], userId: number) => {
+  const foundUser = users.find(user => user.id === userId);
+
+  return foundUser || null;
+};
+
+const extendedTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUserById(usersFromServer, todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<ExtendedTodo[]>(extendedTodos);
+
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const makeTodoId = Math.max(...todos.map(todo => todo.id)) + 1;
```

Comment:
Better move it inside handleFormSubmit to prevent recalculation every time your component renders.

Reply:
@SerhiiRomaniukWD You didn't fix this. Can you comment?

Reply:
@sergii-nosachenko O sry I didn't see this comment
I will fix it right now